Fix: getopt.h:147:12: error: conflicting types for 'getopt'

This fixes building on macOS 27 Golden Gate
===================================================================
diff -urN src/getopt.h.orig src/getopt.h
--- src/getopt.h.orig	2025-01-13 16:49:26.000000000 -0500
+++ src/getopt.h	2026-10-26 02:14:13.526327536 -0400
@@ -138,7 +138,7 @@
    `getopt'.  */

 #if (defined __STDC__ && __STDC__) || defined __cplusplus
-# ifdef __GNU_LIBRARY__
+# if defined(__GNU_LIBRARY__) || defined(__APPLE__)
 /* Many other libraries have conflicting prototypes for getopt, with
    differences in the consts, in stdlib.h.  To avoid compilation
    errors, only prototype getopt for the GNU C library.  */
